content/developer/reference/backend/actions.rst

```diff
@@ -442,34 +442,90 @@ Actions triggered automatically on a predefined frequency.
     Priority of the action when executing multiple actions at the same time
 
 
-Advanced use: Batching
+Writing CRON functions
 ----------------------
 
-When executing a scheduled action, it's recommended to try batching progress in order
-to avoid hogging a worker for a long period of time and possibly running into timeout exceptions.
+When running a scheduled action, it's recommended that you try to batch the
+progress in order to a worker for a long period of time and possibly run into
+timeout exceptions. Therefore you should split the processing so that each call
+make progress on some of the work to be done.
 
-Odoo provides a simple API for scheduled action batching;
+When writing a such a function, you should focus on processing a single batch.
+A batch should process one or many records and should generally take no more
+than *a few seconds*.
 
 .. code-block:: python
 
-      self.env['ir.cron']._notify_progress(done=XX:int, remaining=XX:int)
+    def _cron_do_something(self, *, limit=300):  # limit: allows for tweaking
+        domain = [('state', '=', 'ready')]
+        records = self.search(domain, limit=limit)
+        records.do_something()
+        # notify progression
+        remaining = 0 if len(records) == limit else self.search_count(domain)
+        self.env['ir.cron']._commit_progress(len(records), remaining=remaining)
+
+Work is committed by the framework after each batch. The framework will
+call the function as many times as necessary to process the remaining work.
+
+If called from outside of the cron job, the progress function call will have
+no effect.
+
+There are some cases, where you may want to keep resources between multiple
+batches or want to loop yourself to handle exceptions.
+If this is the case, you must notify the scheduler of the progress of your work
+by calling ``ir.cron._commit_progress`` and checking the result. The progress
+function returns the number of seconds remaining for the call; if it is 0, you
+must return as soon as possible.
```

Comment:
What do you mean by "as soon as possible"? Not to be nitpicky, but it could be interpreted in widely different ways. "immediately"? "without any new DB access"?

Reply:
It is intentionally left open to interpretation. At some point, you will reach the hard limit and be killed.

Example: Some jobs may do a logical batch consisting of : loop(A, commit, B, commit). When after A they have no time left, they may decide to still do B to have consistent results, but they should not continue looping. If they want, they may continue  running at he risk of being killed and unfair to other jobs.